### PR TITLE
feat(gateway): add error reporting via Sentry

### LINKED ISSUE #7099: Sentry integration with gateways
```[tasklist]
### DoD
- [ ] https://github.com/firezone/firezone/pull/7103
- [x] Documentation update?
- [x] Do we need to update the "Deploy" page in the portal to mention telemetry?
```

### PR DESCRIPTION
**Pull request by thomaseizinger**
Similar to the GUI and headless clients, adding error reporting via Sentry should give us much better insight into how well gateways are performing.

Resolves: #7099.

### CODE REVIEW

File: rust/telemetry/src/lib.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@ pub struct Dsn(&'static str);
 // > DSNs are safe to keep public because they only allow submission of new events and related event data; they do not allow read access to any information.
 // <https://docs.sentry.io/concepts/key-terms/dsn-explainer/#dsn-utilization>
 
+pub const GATEWAY_DSN: Dsn = Dsn("TODO");
```
Comment (thomaseizinger) [inline rust/telemetry/src/lib.rs]: I don't have permissions to create a new project so I opened #7102.

File: website/src/components/Changelog/Gateway.tsx [hunk 2]
```diff
@@ -19,6 +19,9 @@ export default function Gateway() {
             Implements support for the new control protocol; delivering faster
             and more robust connection establishment.
           </ChangeItem>
+          <ChangeItem pull="6941">
```
Comment (thomaseizinger) [inline website/src/components/Changelog/Gateway.tsx]:
```suggestion
          <ChangeItem pull="7103">
```

File: rust/gateway/src/main.rs [hunk 3]
```diff
@@ -36,19 +37,27 @@ async fn main() {
         .install_default()
         .expect("Calling `install_default` only once per process should always succeed");
 
+    let cli = Cli::parse();
+    let telemetry = Telemetry::default();
+    telemetry.start(
+        cli.api_url.as_str(),
+        firezone_bin_shared::git_version!("gateway-*"),
+        firezone_telemetry::GATEWAY_DSN,
+    );
```
Comment (thomaseizinger) [inline rust/gateway/src/main.rs]: It would be pretty easy to add a CLI toggle to disable this. Should we?

**Review by jamilbk — COMMENTED** [review]
Could you also update the connectivity table in the gateway deploy docs at /kb/deploy/gateways?

**Review by ReactorScram — APPROVED** [review]

File: website/src/app/kb/deploy/gateways/readme.mdx [hunk 4]
```diff
@@ -156,6 +157,12 @@ It's a good idea to keep your Gateways up to date with the latest version
 available. See [upgrading Gateways](/kb/administer/upgrading) for ways to
 automate this.
 
+## Telemetry
+
+By default, gateways will run a https://sentry.io crash-reporting agent. If
```
Comment (jamilbk) [inline website/src/app/kb/deploy/gateways/readme.mdx]:
```suggestion
By default, Gateways will run a https://sentry.io crash-reporting agent. If
```

**Review by jamilbk — APPROVED** [review]

**Pull request merged**